### ISSUE TITLE
Order Editing: Enable optimistic update for Shipping Details and Billing Address

### ISSUE THREAD

**Issue by anas-p** [woocommerce/woocommerce-ios#5610]
Subtask of #5339: Order Editing: Enable optimistic updates for order editing actions.

## Description
Implement optimistic updates to edit, order Shipping Address and Billing Address to improve the experience within the Order details flow.

## Existing

https://user-images.githubusercontent.com/19687299/144557488-f7d9a1e9-1be9-4e29-baef-772a1746f250.MP4

## Expected Behaviour

There should not be any delay in displaying the changes after tapping done.

## To Reproduce:
Steps to reproduce the behaviour:

**Shipping Address:**
1. Open the Orders tab
2. Tap on any order
3. Tap the edit button in Shipping Details
4. Update shipping Details
5. Tap "Done"

**Billing Address:**
1. Open the Orders tab
2. Tap on any order
3. Tap View Billing Information
4. Tap the edit button in Billing Address
4. Update billing address
5. Tap "Done"

**Comment by Juanpe** [comment]
Depends on #6674

**Comment by Juanpe** [comment]
Hi! 👋🏼

I'm going to split it into some subtasks for ease of development and review:

- [ ] #6724
- [ ] #6727
- [ ] #6728

I'll update this comment during the development.

Thanks!

CC: @itsmeichigo 🙂

**Comment by JorgeMucientes** [comment]
Closing as part of the Woo Mobile GitHub stale-issue audit cleanup.

Rationale for closing:
- Listed in audit section "Potentially Stale / Good Candidates To Close Still Open, Outside Grouped Commerce Rows (40)". Last GitHub activity in the audit: 2022-04-27 1527d. Audit recommendation: Keep / Manual review. Audit rationale: Open in the 2026-07-02 live snapshot; last GitHub update 2022-04-27 (1527d ago). No direct current-code signal found; manual product triage recommended.. No longer in a protected grouped area in this report; close as stale backlog cleanup and reopen if still reproducible or product-owned.

Reopen if this is still reproducible, product-owned, or should be tracked in a current Linear issue.
